File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
import sys
input = sys.stdin.readline

def func():
    start, mid, end = '', '', ''

    if len(words) == 1:
        return words[0]
    
    while len(words) > 1:
        x = words.pop()
        y = words.pop()

        if x == y:
            start = x + start
            end = end + y

        else:
            if len(words) == 0:
                return "I'm Sorry Hansoo"
            z = words.pop()
            if z == y:
                mid += x
                start = y + start
                end = end + z
            else:
                return "I'm Sorry Hansoo"
            
    if words:
        mid += words[0]

    if len(mid) > 1:
        return "I'm Sorry Hansoo"

    return start + mid + end


if __name__ == '__main__':
    input_words = list(input().strip())
    words = sorted(input_words)
    print(func())

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.js
// 알고리즘 로직을 처리하는 함수
function make_pal(words){
    let keys = Object.keys(words).sort();
    let [left, mid, right] = ['', '', ''];
    for (let key of keys){
        if (words[key] % 2 === 0){
            for (let i = 0; i < words[key] / 2; i++){
                left = left + key
                right = key + right
            }
        }else{
            if (mid === ''){
                mid += key

                for (let i = 0; i < Math.floor(words[key] / 2); i++){
                    left = left + key
                    right = key + right
                }
            }else{
                return "I'm Sorry Hansoo"
            }
        }
    }
    return left + mid + right
}


function solve(input) {
  // TODO: 알고리즘 로직
    let words = input[0];
    let cnt = {};
    for (let word of words){
        if (cnt[word]){
            cnt[word] += 1;
        }
        else{
            cnt[word] = 1;
        }
    }
    console.log(make_pal(cnt));
}

// 제출하기전에 false 로
const isLocal = false;  // 여기서 true로 설정하면 로컬 환경으로 간주

if (isLocal) {
    // 로컬 환경 (VSCode)에서의 입력 처리
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    let lines = [];
    rl.on('line', (line) => {
        lines.push(line);
    }).on('close', () => {
        solve(lines);
    });
} else {
    // 백준 환경에서의 입력 처리
    const fs = require('fs');
    const lines = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
    solve(lines);
}